File: leetcode-python-go/1814. 统计一个数组中好对子的数目/main.go
package main

import (
	"fmt"
)

func countNicePairs(nums []int) (ans int) {
	cnt := map[int]int{}
	for _, num := range nums {
		rev := 0
		for x := num; x > 0; x /= 10 {
			rev = rev*10 + x%10
		}
		ans += cnt[num-rev]
		cnt[num-rev]++
	}
	return ans % (1e9 + 7)
}

func main() {
	nums := []int{42, 11, 1, 97}
	results := countNicePairs(nums)
	fmt.Println(results)
}

File: leetcode-python-go/1814. 统计一个数组中好对子的数目/main.py
from typing import List
from collections import Counter

nums = [13,10,35,24,76]

def countNicePairs(nums: List[int]) -> int:
    res = 0
    cnt = Counter()
    for i in nums:
        j = int(str(i)[::-1])
        res += cnt[i - j]
        cnt[i - j] += 1
    return res % (10 ** 9 + 7)

print(countNicePairs(nums))